zip-0213.rst

```diff
@@ -0,0 +1,179 @@
+  recovered using a 32-byte array of zeroes as the outgoing viewing key.
+
+
+Interaction with funding streams
+--------------------------------
+
+If ZIP 207 [#zip-0207]_ is accepted and deployed prior to (or in the same network upgrade
+as) this ZIP, the following additional consensus rules are enforced once the TODO network
+upgrade activates:
+
+- Funding streams MAY target Sapling addresses in addition to transparent P2SH addresses.
+
+- The existing consensus rule from ZIP 207 [#zip-0207]_ requiring coinbase transactions
+  for block height ``height`` to pay active funding streams with a standard P2SH script,
+  is amended to only apply to active funding streams that target a transparent address at
+  ``height``.
+
+- In the coinbase transaction for block height ``height``, for every active funding stream
+  that targets a Sapling address at ``height``, the transaction MUST include at least one
+  Sapling output that pays exactly the funding stream's value to the target Sapling
+  address.
+
+
+Interaction with the Founders' Reward
+-------------------------------------
+
+This ZIP does not alter the existing Founders' Reward addresses. If a future ZIP proposed
+moving to shielded Founders' Reward addresses (instead of, or before, deploying ZIP 207),
+the necessary consensus rules would be similar to those specified above for funding
+streams.
+
+
+Rationale
+=========
+
+The ZIP does not require that all coinbase must be shielded immediately from activation of
+the network upgrade, so that miners and mining pools may gradually migrate from their
+existing transparent addresses to Sapling addresses. This also simplifies the consensus
+rules, because the Founders' Reward targets transparent addresses, and thus it remains
+necessary for the time being to support them. A future ZIP could require all coinbase to
+be shielded immediately.
+
+Enforcing coinbase maturity at the consensus level for Sapling outputs would incur
+significant complexity in the consensus rules, because it would require special-casing
+coinbase note commitments in the Sapling commitment tree. The standard recommendation when
+spending a note is to select an anchor 10 blocks back from the current chain tip, which
+acts as a de-facto 10-block maturity on all notes, coinbase included. This might be
+proposed as a consensus rule in future.
+
+There is another reason for shielded coinbase maturity being unnecessary: when a rollback
+occurs that would cause a shielded coinbase output to disappear, it will also invalidate
+every shielded transaction that uses an anchor descending from the tree that the shielded
+coinbase output had been appended to. That is, all economic activity would be rolled back
+in addition to the shielded coinbase output disappearing, so there is no reason to make
+shielded coinbase a special case when the same behaviour occurs in regular shielded notes
+already. In the transparent coinbase case, only direct child transactions of the
```

Comment:
I got a bit lost here. It would help to explain why coinbase maturity rules exist in the first place, and why the above reasoning that they are unnecessary does not apply to transparent coinbase outputs. I can't see why that's the case.

Reply:
Agreed. I'm unsure what side-effects there might be to making it possible to spend shielded coinbases right away. A couple things come to mind:

- Change in economic incentives: miners can sell their earnings ~4 hours sooner, potentially enabling them to profit from attacks (e.g. a 51% attack) they believed would not have been profitable before because they expected the attack to be detected and the price to crash before their mining rewards matured.
- A malicious miner has has a new resource: the rewards from blocks they mine become available immediately. That might increase the strength of already-known attacks. If there's any situation where a malicious miner needs to send bribes while they are carrying out an attack, being able to use the block rewards earned as part of the attack to bribe would reduce the up-front cost of the attack. For example maybe they're renting GPUs to mine a competing chain and as part of that attack they immediately send their block rewards to miners' of other chains to incentivize them to start mining on the shorter chain.

Is there anything else I'm missing?